services/autoscaling/src/simcore_service_autoscaling/utils/utils_docker.py

```diff
@@ -81,6 +80,60 @@ def _check_if_node_is_removable(node: Node) -> bool:
     return nodes_that_need_removal
 
 
+_DISALLOWED_DOCKER_PLACEMENT_CONSTRAINTS: Final[list[str]] = [
+    "node.id",
+    "node.hostname",
+    "node.role",
+]
+
+_PENDING_DOCKER_TASK_MESSAGE: Final[str] = "pending task scheduling"
+_INSUFFICIENT_RESOURCES_DOCKER_TASK_ERR: Final[str] = "insufficient resources on"
+
+
+def _is_task_waiting_for_resources(task: Task) -> bool:
+    # NOTE: https://docs.docker.com/engine/swarm/how-swarm-mode-works/swarm-task-states/
+    if (
+        not task.Status
+        or not task.Status.State
+        or not task.Status.Message
+        or not task.Status.Err
+    ):
+        return False
+    return (
+        task.Status.State == TaskState.pending
+        and task.Status.Message == _PENDING_DOCKER_TASK_MESSAGE
```

Comment:
are we sure this is always the message? check state on messages (specially exact checks) is always dangerous

Reply:
well I'm happy if you find an alternative. anyway this PR just changed this to a variable.